server/routes/api/templates.js

```diff
@@ -0,0 +1,151 @@
+router.post( "/",
+    passport.authenticate("jwt", { session: false }),
+    async (req, res) => {
+        const ownedBy = req.user.id;
+        try{
+            const { errors, isValid } = validateTemplateInput(req.body);
+            if (!isValid) {
+                return res.status(400).json(errors);
+            }
+            const { name, subject, content, attachments } = req.body;
+
+            const existingTemplate = await Template.find({
+                name: name,
+                ownedBy: ownedBy
+            });
+            if(existingTemplate && existingTemplate.length > 0){
+                res.status(409).send({error: "Template with this title already exists"});
+                return;
+            }
+            const template = new Template({ name, subject, content, ownedBy, attachments});
+            const newTemplate = await template.save();
+            res.status(200).send(newTemplate);
+        } catch(errors) {
+            console.log(errors);
+            res.status(400).send({error : "Error creating template"});
+        }
+});
+
+// @route GET /api/templates
+// @desc Get all templates owned by user
+// @access Authenticated Users
+router.get("/",
+    passport.authenticate("jwt", { session: false }),
+    async (req, res) => {
+        const ownedBy = req.user.id;
+        try{
+            const templates = await Template.find({ ownedBy: ownedBy });
+            res.status(200).json(templates);
+        } catch (errors) {
+            console.log(errors);
+            res.status(400).send({error : "Error retrieving templates"});
+        }
+});
+
+// @route GET /api/templates/:id
+// @desc Get Template by id
+// @access Authenticated Users
+router.get("/:id",
+    passport.authenticate("jwt", { session: false }),
+    async (req, res) => {
+        
+        const templateId = req.params.id;
+        const ownedBy = req.user.id;
+
+        try {
+            const template = await Template.find({
+                _id: templateId,
+                ownedBy: ownedBy
+            });
+            if (!template || template.length === 0) {
+                res
+                    .status(404)
+                    .send({ id: `Template with id ${templateId} is not found` });
+            } else {
+                res.status(200).json(template[0]);
+            }
+        } catch (errors) {
+            console.log(errors);
+            res.status(400).send({error : "Error retrieving template"});
+        }
+        
+});
+
+// @route PUT /api/templates
+// @desc Modify template by template id
+// @access Authenticated Users
+router.put("/:id",
+    passport.authenticate("jwt", { session: false }),
+    async (req, res) => {
+        console.log("Got to the put")
+        const templateId = req.params.id;
+        const ownedBy = req.user.id;
+
+        const template = await Template.find( {
+            _id: templateId,
+            ownedBy: ownedBy
+        });
+        console.log("Got to the put 2")
```

Comment:
lol you should remove these console logs before we merge

Reply:
dang